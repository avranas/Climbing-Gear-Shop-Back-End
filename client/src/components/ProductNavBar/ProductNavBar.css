#product-navigation-list {
  padding: 1rem;
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: flex;
  justify-content: space-between;
}

#product-navigation-list ul {
  margin: 0;
}

#product-navigation-list li {
  list-style: none;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
  font-weight: 100;
}

#product-navigation-list a:visited,
#product-navigation-list a:link {
  text-decoration: none;
  color: #111111;
}

#product-navigation-list a:hover {
  text-decoration: underline;
}

#product-navigation-list a:active {
  border-style: solid;
  border-color: blue;
  border-width: 0.15rem;
}

@media only screen and (max-width: 770px) {
  #product-navigation-list li {
    font-size: 0.8rem;
  }
}

@media only screen and (max-width: 512px) {
  #product-navigation-list li {
    font-size: 0.6rem;
  }
  #product-navigation-list {
    padding: 0.5rem;
  }
}

@media only screen and (max-width: 350px) {
  #product-navigation-list {
    display: none;
  }
}
