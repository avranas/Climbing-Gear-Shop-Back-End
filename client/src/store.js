import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});