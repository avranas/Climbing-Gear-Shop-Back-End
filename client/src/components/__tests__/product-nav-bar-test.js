import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import ProductNavBar from "../ProductNavBar/ProductNavBar";

afterEach(() => {
  cleanup();
});

const productNavBar = (
  <Provider store={store}>
    <BrowserRouter>
      <ProductNavBar />
    </BrowserRouter>
  </Provider>
);

test("ProductNavBar matches snapshot", async () => {
  render(productNavBar);
  const tree = renderer.create(productNavBar).toJSON();
  expect(tree).toMatchSnapshot();
});
