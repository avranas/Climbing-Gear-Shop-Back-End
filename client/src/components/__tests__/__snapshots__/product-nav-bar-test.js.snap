// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNavBar matches snapshot 1`] = `
<nav
  className="container styled-box"
  id="product-navigation-bar"
>
  <ul
    id="product-navigation-list"
  >
    <li>
      <a
        href="/products/0"
        onClick={[Function]}
      >
        All Products
      </a>
    </li>
    <li>
      <a
        href="/products/0?category=shoes"
        onClick={[Function]}
      >
        Shoes
      </a>
    </li>
    <li>
      <a
        href="/products/0?category=harnesses"
        onClick={[Function]}
      >
        Harnesses
      </a>
    </li>
    <li>
      <a
        href="/products/0?category=ropes"
        onClick={[Function]}
      >
        Ropes
      </a>
    </li>
    <li>
      <a
        href="/products/0?category=hardware"
        onClick={[Function]}
      >
        Hardware
      </a>
    </li>
    <li>
      <a
        href="/products/0?category=chalk"
        onClick={[Function]}
      >
        Chalk
      </a>
    </li>
  </ul>
</nav>
`;
